mne/utils/check.py

```diff
@@ -420,15 +420,21 @@ def _check_rank(rank):
 
 def _check_one_ch_type(info, picks, noise_cov, method):
     """Check number of sensor types and presence of noise covariance matrix."""
+    from ..cov import make_ad_hoc_cov, Covariance
     from ..io.pick import pick_info
     from ..channels.channels import _contains_ch_type
     info_pick = pick_info(info, sel=picks)
     ch_types =\
         [_contains_ch_type(info_pick, tt) for tt in ('mag', 'grad', 'eeg')]
-    if method == 'lcmv' and sum(ch_types) > 1 and noise_cov is None:
-        raise ValueError('Source reconstruction with several sensor types '
-                         'requires a noise covariance matrix to be '
-                         'able to apply whitening.')
-    elif method == 'dics' and sum(ch_types) > 1:
-        warn('The use of several sensor types with the DICS beamformer is '
-             'not heavily tested yet.')
+    if sum(ch_types) > 1:
+        if method == 'lcmv' and noise_cov is None:
+            raise ValueError('Source reconstruction with several sensor types'
+                             ' requires a noise covariance matrix to be '
+                             'able to apply whitening.')
+        if method == 'dics':
+            warn('The use of several sensor types with the DICS beamformer is '
+                 'not heavily tested yet.')
```

Comment:
Should we convert this into an error message? We discussed this in #5882 but never reached a conclusion. I think results from this are likely wrong.

Reply:
Done